src/ci/shared.sh

```diff
@@ -15,6 +15,12 @@
 
 # See http://unix.stackexchange.com/questions/82598
 function retry {
+  # Only do retries on CI, locally they shouldn't really be necessary
+  if [ -z "${CI:-}" ]; then
+      "$@"
+      return
+  fi
+
```

Comment:
Just to expand on my reluctance - one reason we do CI is so we know other people can build it correctly. I can imagine myself getting frustrated by an intermittent failure, just to realise that rust uses a magic variable to dodge it.

Is there a particular reason for changing it as part of this PR?